### ISSUE TITLE
Feature: Doc template generator

### ISSUE THREAD

**Issue by llogiq** [getsynth/synth#31]
**Required Functionality**
We feel that for now, synth namespaces are not currently well documented. Some markdown (or similar format) describing the meaning of columns, etc. would really go a long way to make the definitions more approachable.

**Proposed Solution**
To reduce the work to set up such documentation and induce a certain consistency, a `synth doc` subcommand could parse the schema and issue a README.md in the namespace that can then be extended to create some usable documentation.

The template should have the following sections:

* A short description (obviously left as an exercise to the author)
* The collections of the namespace with
   * field names, types and descriptions (the latter also to be added by the author)
   * dependencies to other collections (e.g. foreign keys, subcollections)
* A minimal pretty-printed (perhaps abridged if too long) generated output of the namespace.

**Use case**
The documentation would help with maintaining the schemas, especially if the original author is no longer around. Making it easy to document means we'll hopefully see better documented schemas.

**Comment by OLUWAMUYIWA** [comment]
In generating the .md, I want to make use of another crate, the only one I saw in crates.io is `markdown-gen`. Is it okay to add another dependency?

**Comment by christoshadjiaslanis** [comment]
> In generating the .md, I want to make use of another crate, the only one I saw in crates.io is `markdown-gen`. Is it okay to add another dependency?

Yeah I think it makes a lot of sense to not have to re-invent the wheel when it comes to generating semantically correct markdown. This [crate](https://crates.io/crates/md_gen) may also work.
